src/gui/resourcetreeitem.h

```diff
@@ -56,24 +56,36 @@ struct Archive {
     uint32 index;
 };
 
-struct ItemData {
-    ItemData(const QString &parentPath, const QString &fileName, Aurora::Archive *archiveData, Aurora::Archive::Resource &resource);
-    ItemData(const QString &fullPath, const QFileInfo &info);
+struct SoundData {
+    mutable bool triedDuration;
+    mutable uint64 duration;
+};
+
+struct FSData {
+    FSData(const QString &parentPath, const QString &fileName, Aurora::Archive *archiveData, Aurora::Archive::Resource &resource);
+    FSData(const QString &fullPath, const QFileInfo &info);
 
     QString _fullPath; // used for QFileInfo and item info e.g. logging
     bool _isDir;
-    Source _source;
-    Aurora::FileType _fileType;
-    Aurora::ResourceType _resourceType;
     qint64 _size;
+};
+
+class ResourceTreeItem {
+private:
+    ResourceTreeItem *_parent;
+    QList<ResourceTreeItem*> _children;
+    QString _name; // The filename. It's what the tree view displays.
```

Comment:
This comment here, on the other hand, looks like it should be a Doxygen comment. Use `///<` to create a Doxygen comment that points to the preceeding item on the same line.

Like so:
```
QString _name; ///< The filename. It's what the tree view displays.
```